client/dashboard/app/layout.tsx

```diff
@@ -47,13 +57,15 @@ function AnalyticsProviderWithClient( { children }: { children: React.ReactNode
 }
 
 function Layout( { config }: { config: AppConfig } ) {
+	const router = useMemo( () => getRouter( config ), [ config ] );
+
 	return (
 		<AppProvider config={ config }>
 			<QueryClientProvider client={ queryClient }>
 				<AuthProvider>
 					<I18nProvider>
-						<AnalyticsProviderWithClient>
-							<RouterProviderWithAuth config={ config } />
+						<AnalyticsProviderWithClient router={ router }>
```

Comment:
It seems we should use the [Wrap](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType#wrap-property) option in createRouter to access the site based on siteSlug directly within AnalyticsProviderWithClient. 🤔

Reply:
Oops, it seems we still need the getGetSuperProps function. However, we could make it a reference to the latest function, or use a ref to store the most recently selected site and then use that within the function.

That said, combining it with the site-specific props inside the analytics client functions could definitely simplify things 🤔